packages/roosterjs-editor-api/lib/format/toggleBlockQuote.ts

```diff
@@ -0,0 +1,119 @@
+                nodes = blockElement.getContentNodes();
+                startContainer = range.startContainer;
+                startOffset = range.startOffset;
+
+                // If the only content node is <BR>, we wrap it with <DIV>, otherwise the format will break. This often happens in firefox.
+                if (nodes.length == 1 && nodes[0].nodeName == 'BR') {
+                    nodes[0] = wrap(nodes[0], '<div></div>') as HTMLDivElement;
+                    startContainer = nodes[0];
+                    startOffset = NodeBoundary.Begin;
+                }
+            } else { // Selection is collapsed and blockElment is null, we need to create an empty div. In case of IE and Edge, we insert ZWS to put cursor in the div, otherwise insert BR node
+                let div = editor.getDocument().createElement('div');
+
+                if (browserData.isEdge || browserData.isIE) {
+                    div.innerHTML = ZERO_WIDTH_SPACE;
+                } else {
+                    let brNode = editor.getDocument().createElement('br');
+                    div.appendChild(brNode);
+                }
+                editor.insertNode(div, {
+                    position: ContentPosition.SelectionStart,
+                    updateCursor: true,
+                    replaceSelection: true,
+                    insertOnNewLine: false,
+                });
+                startContainer = div.firstChild;
+                startOffset = NodeBoundary.Begin;
+                nodes = [div];
+            }
+
+            let quoteElement = wrapAll(nodes, '<blockquote></blockqupte>') as HTMLQuoteElement;
+            if (!!styler) {
+                styler(quoteElement);
+            }
+            updateSelection(range, editor, startContainer, startOffset, isRangeCollapsed);
+        };
+
+        if (formatter) {
+            execFormatWithUndo(editor, formatter);
+        }
+    } else { // Current selection is in blockquote, need to unblockquote the selection
+        let blockQuoteElements: Node[] = [];
+
+        // Selection may contain multiple blockquotes, check and unblockquote all
+        while (blockElement) {
+            let containerNode = blockElement.getStartNode();
+            let blockQuoteElement = getBlockQuoteElement(editor, containerNode);
+            if (blockQuoteElement && blockQuoteElements.indexOf(blockQuoteElement) == -1) {
+                blockQuoteElements.push(blockQuoteElement);
+            }
+            blockElement = contentTraverser.getNextBlockElement();
+        }
+
+        let formatter = () => {
+            let startContainer = range.startContainer;
+            let startOffset = range.startOffset;
+            for (let element of blockQuoteElements) {
+                unwrap(element);
+            }
+            updateSelection(range, editor, startContainer, startOffset);
+        };
+        if (formatter) {
```

Comment:
same question, you declare formatter yourselft, why do you need to double check that formatter is not null?

Reply:
Removed